### ISSUE TITLE
Performance: reduce log spam

### ISSUE THREAD

**Issue by JonnyOThan** [KSPModdingLibs/Shabby#18]
Shabby currently emits a log line every time it does a shader replacement.  We should make those conditionally compiled for performance.

**Comment by Rodg88** [comment]
It also logs each time it fails due to shaders not being found, which currently happens a fair bit if someone has a mod using Resurfaced without Resurfaced installed.

This particular case might be better taken care of with MM conditional patching instead though, as you probably want errors to still show in the log for actual troubleshooting?

So for eg instead of what BDB currently does (just adding the `SHABBY_MATERIAL_REPLACE` block into each part cfg without MM patching/conditionals), we could just add a `:NEEDS[Resurfaced]` to the `SHABBY_MATERIAL_REPLACE`. Would this be faster in terms of load times?

Afaik, it was set up to only have `SHABBY_MATERIAL_REPLACE` without any MM handling for 'simplicity', but it's not really much more complicated to add a NEEDS in there, esp if it speeds loading in some cases.

**Comment by JonnyOThan** [comment]
Yeah, emitting an error when something goes wrong is fine; but Shabby is currently emitting log messages for every single successful replacement too.

**Comment by drewcassidy** [comment]
yup, not worried about performance when something is broken anyways

It should be faster now, not sure how I would measure this for it to be anything more than statistical noise

**Comment by JonnyOThan** [comment]
@Rodg88 
> So for eg instead of what BDB currently does (just adding the SHABBY_MATERIAL_REPLACE block into each part cfg without MM patching/conditionals), we could just add a :NEEDS[Resurfaced] to the SHABBY_MATERIAL_REPLACE. Would this be faster in terms of load times?

Yes, because once all the MM patches are cached the ones with :NEEDS will be completely removed and act like they were never there in the first place.
